packages/node/src/plugins/webpackChunkUtilities.ts

```diff
@@ -280,42 +280,10 @@ export function generateLoadScript(runtimeTemplate: any): string {
             if (!name) {
               throw new Error('__webpack_require__.l name is required for ' + url);
             }
-            var remoteName = name;
-            if(name.includes('__remote_scope__')) {
-              remoteName = name.split('__remote_scope__.')[1]
-            }
-            if (typeof globalThis.__remote_scope__[remoteName] !== 'undefined') return callback(globalThis.__remote_scope__[remoteName]);
-            globalThis.__remote_scope__._config[remoteName] = url;
-            try {
-              const scriptContent = await (globalThis.webpackChunkLoad || globalThis.fetch || require("node-fetch"))(url).then(res => res.text());
-              let remote;
-              if (typeof process !== 'undefined') {
-                const vm = require('vm');
-                const m = require('module');
-                const remoteCapsule = vm.runInThisContext(m.wrap(scriptContent), 'node-federation-loader-' + name + '.vm')
-                const exp = {};
-                remote = {exports:{}};
-                remoteCapsule(exp,require,remote,'node-federation-loader-' + name + '.vm',__dirname);
-                remote = remote.exports || remote;
-              } else {
-                remote = eval('let module = {};' + scriptContent + '\\nmodule.exports')
-              }
-              globalThis.__remote_scope__[remoteName] = remote[remoteName] || remote;
-              globalThis.__remote_scope__._config[remoteName] = url;
-              callback(globalThis.__remote_scope__[remoteName])
-            } catch (e) {
-              e.target = {src: url};
-              globalThis.__remote_scope__[remoteName] = {
-                get: function() {
-                  return function() {
-                    return ()=>null
-                  }
-                },
-                init: function() {},
-                fake: true
-              }
-              callback(e);
-            }
+            return ${RuntimeGlobals.require}.federation.runtime.loadScriptNode(url, {attrs: {}}).then(function(res){
+            globalThis[name] = res[name] || res;
```

Comment:
@2heal1 becuase we are pretending this is script type, webpack will always look on global scope based on @ syntax

in the past i used globalThis.remoteScope.remoteName@http:// so that i could get module from object and not pollute globals. 

im wondering if we can do similar here but to require.federation.instance.moduleCache.get()?

Reply:
if the remote use @ syntax , it will be collected to initOptions.remotes . And it can automaticlly be loaded by runtime , the corresponding code is [here](https://github.com/module-federation/universe/blob/bcc1be9e8140f0c5fa8e517ca31bb50729bd614d/packages/webpack-bundler-runtime/src/remotes.ts#L122).

Reply:
ahh i check the code again , find it has some coupled with prev implementation , so it may not run as i thought . 

But yes, we can try to get the loadedModule by require.federation.instance.moduleCache.get() if the module has been loaded by runtime

Reply:
ahh clever! ill try searching module cache, then global, then will load. Great idea.

Reply:
Yes, I think we would need to change the external script type module that webpack uses, which implements a callback to global. I will try and see if I can append `moduleCache.get()` to it as "name" and see if it works.